pkg/features/features.go

```diff
@@ -106,7 +107,7 @@ var allFeatureGates = map[featuregate.Feature]featuregate.FeatureSpec{
 	ManagedIstio:       {Default: true, PreRelease: featuregate.Deprecated},
 	APIServerSNI:       {Default: true, PreRelease: featuregate.Deprecated},
 	SeedChange:         {Default: true, PreRelease: featuregate.Beta},
-	ReversedVPN:        {Default: true, PreRelease: featuregate.Beta},
+	ReversedVPN:        {Default: true, PreRelease: featuregate.Deprecated},
```

Comment:
I think the idea is not to deprecate it but to promote it to GA and to lock it to its default (`true`). Afterwards, in future releases the feature gate can be cleaned up.

```suggestion
	ReversedVPN:        {Default: true, PreRelease: featuregate.GA, LockToDefault: true},
```

Reply:
As all code for the legacy VPN is removed with this change, I would consider the feature flag as deprecated.
Making it GA and locking it to a default value seems to me inconsistent.
Making it deprecated results in a warning during gardenlet startup. I would guess, that's something we want to have.
However, locking it to the default, results in gardelet crashing, if someone sets it to false in the gardenlet config.
My feeling is that, it is better to just print a warning and to continue, than crashing in this case.

Reply:
I cannot agree. 
Let me elaborate my understanding of feature gates and their lifecycle. In 90% of the cases the lifecycle of a feature gate is:
1. Gets introduced as alpha (usually disabled by default)
2. Over time gets promoted to beta (usually gets enabled by default)
3. When we are confident with the feature, gets promoted to GA (and also usually gets locked to its default - enabled by default to make sure that everyone is using the feature and we can safely clean it up in few releases).
4. Feature gate gets removed in few releases after step 3.

With regards to the deprecation, I think that it should be used only when we discontinue the feature - there is better alternative; we don't need the feature; etc. IMO, examples of proper feature gate deprecations in gardener:
- https://github.com/gardener/gardener/pull/6599 - we introduced another way of achieving the same feature via the Shoot spec and we no longer needed the old feature
- https://github.com/gardener/gardener/pull/5985 - the feature was asymmetric to the other other end-user credentials

The semantic I understand behind deprecated is "the feature gate won't be developed anymore, there is probably a better alternative of achieving the same thing, stop using this feature". 

The `ReversedVPN` is actually an example of feature gate that was introduced and then graduated over time. It should fall in these 90% of the cases because that is semantically the correct way. We believe the feature is GA and we enable it by default.

If you are looking into the ManagedIstio and APIServerSNI feature gates as examples - I don't believe that deprecating them was the right thing semantically. Because deprecated means "don't use the feature anymore" but today you cannot even create an AWS Shoot when these features are disabled (ref https://github.com/gardener/gardener-extension-provider-aws/issues/536).

Reply:
@ialidzhikov thanks for that explanation! I guess, my understanding of feature gates was wrong.
According `LockToDefault`, in which case should it be set to `true`? I see that some GA feature have it set and others don't.

Reply:
As you want to drop the old VPN solution in the same PR, then you need LockToDefault=true. Because ReversedVPN=false wouldn't make sense as the old VPN solution will be removed with this change. Hence, you need LockToDefault=true.
If you just want to promote the ReversedVPN to GA but you still want to preserve the old VPN solution and the possibility the feature gate to be disabled, then you don't need LockToDefault=true.

Reply:
Ah, OK. So we have two kind of GA feature gates: Some that can be changed and some that can't. That contradicts to what I learned about [kubernetes  feature gates](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-stages), where GA features can't be disabled.
I guess, there should be a document explaining gardener feature gates, as this was a bit confusing to me.
Anyhow, I'm fine with your suggested change and I'll change it accordingly.